GiniComponents/GiniUtilites/Sources/GiniUtilites/URLOpener.swift

```diff
@@ -22,12 +22,13 @@ public struct URLOpener {
     ///   - completion: called after opening is completed
     ///                 param is true if website was opened successfully
     ///                 param is false if opening failed
-
-    public func openLink(url: URL, completion: ((Bool) -> Void)?) {
+    public func openLink(url: URL, completion: (@MainActor @Sendable (Bool) -> Void)?) {
         if application.canOpenURL(url) {
             application.open(url, options: [:], completionHandler: completion)
         } else {
-            completion?(false)
+            Task { @MainActor in
+                completion?(false)
```

Comment:
We use DispatchQueue.main here as an alternative to @MainActor in Task, which are both part of Swift Concurrency, which is currently available only in iOS 13 and higher. Both should do the same thing - calling the completion block from the main thread.

And since healthSDK supports iOS 12, and Task, @MainActor and other Swift Concurrency things are not available in iOS 12, DispatchQueue.main was used here instead.

I think both solutions are valid, but I think Swift Concurrency should be preferred when possible(it’s more modern, easier, and has better task management), so I added an #available(iOS 13, *) check, and run Task { @MainActor in, both in URLOpener and MockUIApplication of healthSDK.

Reply:
Thank you for explaining that, I totally forgot that we support iOS 12 in Health SDK

Could you please leave a comment about that in the code? like we support iOS 12 or do a check for iOS 12